src/meshes/p4est_mesh_view.jl

```diff
@@ -0,0 +1,209 @@
+                                                           NDIMS)...,
+                                                    prod(trees_per_dimension))
+    calc_tree_node_coordinates!(tree_node_coordinates, parent.nodes, mapping,
+                                trees_per_dimension)
+
+    connectivity = connectivity_structured(trees_per_dimension..., periodicity)
+
+    # TODO: The initial refinement level of 1 should not be hard-coded.
+    p4est = new_p4est(connectivity, 1)
+    p4est_pw = PointerWrapper(p4est)
+
+    # Non-periodic boundaries
+    boundary_names = fill(Symbol("---"), 2 * NDIMS, prod(trees_per_dimension))
+
+    structured_boundary_names!(boundary_names, trees_per_dimension, periodicity)
+
+    return P4estMeshView{NDIMS, eltype(parent.tree_node_coordinates),
+                         typeof(parent.is_parallel),
+                         typeof(p4est_pw), typeof(parent.ghost), NDIMS + 2,
+                         length(parent.nodes)}(PointerWrapper(p4est),
+                                               parent.is_parallel,
+                                               parent.ghost,
+                                               tree_node_coordinates,
+                                               parent.nodes, boundary_names,
+                                               parent.current_filename,
+                                               parent.unsaved_changes,
+                                               parent.p4est_partition_allow_for_coarsening,
+                                               parent, indices_min, indices_max)
+end
+
+@inline Base.ndims(::P4estMeshView{NDIMS}) where {NDIMS} = NDIMS
+@inline Base.real(::P4estMeshView{NDIMS, RealT}) where {NDIMS, RealT} = RealT
+@inline function ntrees(mesh::P4estMeshView)
+    return mesh.p4est.trees.elem_count[]
+end
+@inline ncellsglobal(mesh::P4estMeshView) = Int(mesh.p4est.global_num_quadrants[])
+Base.axes(mesh::P4estMeshView) = map(Base.OneTo, size(mesh))
+Base.axes(mesh::P4estMeshView, i) = Base.OneTo(size(mesh, i))
+Base.size(mesh::P4estMeshView) = size(mesh.tree_node_coordinates)[end]
+function Base.size(mesh::P4estMeshView, i)
+    @unpack indices_min, indices_max = mesh
+    return indices_max[i] - indices_min[i] + 1
+end
+
+function Base.show(io::IO, mesh::P4estMeshView)
+    print(io, "P4estMeshView{", ndims(mesh), ", ", real(mesh), "}")
+end
+
+function Base.show(io::IO, ::MIME"text/plain", mesh::P4estMeshView)
+    if get(io, :compact, false)
+        show(io, mesh)
+    else
+        setup = [
+            "#trees" => ntrees(mesh),
+            "current #cells" => ncellsglobal(mesh),
+            "polydeg" => length(mesh.nodes) - 1,
+        ]
+        summary_box(io,
+                    "P4estMeshView{" * string(ndims(mesh)) * ", " * string(real(mesh)) *
+                    "}", setup)
+    end
+end
+
+function balance!(mesh::P4estMeshView{2}, init_fn = C_NULL)
```

Comment:
Is this routine called for every `P4estViewMesh` instance even though the whole mesh gets actually balanced, right?

Reply:
This should only be called when we call `create_cache`. So, only once during the simulation with no rebalancing. I suppose if we want to rebalance then the cache needs to be created again and `balance!` is called.